### PR TITLE
Update Elasticsearch Documentations

### PR DESCRIPTION
**Pull request by raihankhan**
- Add opensearch quickstart
- Add necessary CRD flowcharts
- Add documentation for latest features

ref - https://github.com/kubedb/project/issues/877
Signed-off-by: raihankhan <raihan@appscode.com>

### CODE REVIEW

File: docs/guides/elasticsearch/README.md [hunk 1]
```diff
@@ -39,8 +39,20 @@ aliases:
...
+## Lifecycle of Elasticsearch Object
+
+<!---
+ref : https://cacoo.com/diagrams/4PxSEzhFdNJRIbIb/0281B
+--->
+<p align="center">
+  <img alt="lifecycle"  src="/docs/guides/elasticsearch/images/Lifecycle-of-an-Elasticsearch-CRD-complete.png">
+</p>
+
+
```
Comment (ArnobKumarSaha) [inline docs/guides/elasticsearch/README.md:52]: Add Autoscaling feature too

File: docs/guides/elasticsearch/concepts/elasticsearch/index.md [hunk 2]
```diff
@@ -29,8 +29,11 @@ metadata:
   name: myes
   namespace: demo
 spec:
+  autoOps:
+    disabled: true
   authSecret:
     name: es-admin-cred
+    externallyManaged: false
```
Comment (ArnobKumarSaha) [inline docs/guides/elasticsearch/concepts/elasticsearch/index.md:36]: externallyManaged should be also be discussed in the below subsection

Comment (ArnobKumarSaha) [reply docs/guides/elasticsearch/concepts/elasticsearch/index.md:36]: Add missing fields for podTemplate & serviceTemplate too

File: docs/guides/elasticsearch/concepts/elasticsearch-ops-request/index.md [hunk 3]
```diff
@@ -81,19 +81,19 @@ It specifies the desired version information required for the Elasticsearch vers
...
+  type: Upgrade
   databaseRef:
-    name: es
+    name: es-quickstart
   upgrade:
-    targetVersion: searchguard-7.5.2-v1
+    targetVersion: xpack-8.5.2
 ```
 
 ### spec.horizontalScaling
```
Comment (ArnobKumarSaha) [inline docs/guides/elasticsearch/concepts/elasticsearch-ops-request/index.md:99]: `spec.configuration` is missing

Comment (ArnobKumarSaha) [reply docs/guides/elasticsearch/concepts/elasticsearch-ops-request/index.md:99]: + `spec.restart` & `spec.timeout`

Comment (ArnobKumarSaha) [reply docs/guides/elasticsearch/concepts/elasticsearch-ops-request/index.md:99]: Add `AppRef` & `tlsSecret` in AppBinding

Comment (ArnobKumarSaha) [reply docs/guides/elasticsearch/concepts/elasticsearch-ops-request/index.md:99]: Add `spec.upgradeConstraints` in ElasticsearchVersion

File: docs/guides/elasticsearch/concepts/elasticsearch/index.md (hunk 2 above)
Comment (ArnobKumarSaha) [reply docs/guides/elasticsearch/concepts/elasticsearch/index.md:36]: Add `opsRequestOptions` field in Elasticsearch Autoscaler.  
Also, have a check on the fields of `autoscaler.spec.compute`

Comment (ArnobKumarSaha) [reply docs/guides/elasticsearch/concepts/elasticsearch/index.md:36]: Where are the docs of `Upgrade`, `Reconfigure`,`ReconfigureTLS`,  `HorizontalScale`, `VerticalScale`, `Restart` opsRequests ?

**Review by ArnobKumarSaha — CHANGES_REQUESTED** [review]

File: docs/guides/elasticsearch/README.md (hunk 1 above)
Comment (raihankhan) [reply docs/guides/elasticsearch/README.md:52]: Added

File: docs/guides/elasticsearch/concepts/elasticsearch/index.md (hunk 2 above)
Comment (raihankhan) [reply docs/guides/elasticsearch/concepts/elasticsearch/index.md:36]: Here are the opsReq documentations: [opsReq](https://github.com/kubedb/docs/blob/elasticseach-doc-update/docs/guides/elasticsearch/concepts/elasticsearch-ops-request/index.md)

Comment (raihankhan) [reply docs/guides/elasticsearch/concepts/elasticsearch/index.md:36]: Added

File: docs/guides/elasticsearch/concepts/elasticsearch-ops-request/index.md (hunk 3 above)
Comment (raihankhan) [reply docs/guides/elasticsearch/concepts/elasticsearch-ops-request/index.md:99]: added

**Review by ArnobKumarSaha — CHANGES_REQUESTED** [review]
Use the global serach on `docs/guides/elasticsearch` folder,  & search for `mongodb`, `mariadb`, `mysql` etc.    There are many. Change them.

**Review by ArnobKumarSaha — DISMISSED** [review]

**Pull request merged**